archive/data_source_archive_file.go

```diff
@@ -220,11 +227,12 @@ func genFileShas(filename string) (string, string, string, error) {
 	h256 := sha256.New()
 	h256.Write([]byte(data))
 	shaSum := h256.Sum(nil)
+	sha256hex := hex.EncodeToString(shaSum[:])
 	sha256base64 := base64.StdEncoding.EncodeToString(shaSum[:])
 
 	md5 := md5.New()
 	md5.Write([]byte(data))
 	md5Sum := hex.EncodeToString(md5.Sum(nil))
 
-	return sha1, sha256base64, md5Sum, nil
+	return sha1, sha256hex, sha256base64, md5Sum, nil
```

Comment:
Might I suggest adding all the missing hash algorithms, per https://github.com/hashicorp/terraform-provider-local/issues/137

E.g.

```go
func genFileShas(filename string) (sha1Hex, sha256Hex, sha256Base64, sha512Hex, sha512Base64, md5Hex string, err error) {
    data, err := ioutil.ReadFile(filename)
    if err != nil {
        err = fmt.Errorf("could not compute file '%s' checksum: %s", filename, err)
        return
    }

    sha1Sum := sha1.Sum(data)
    sha1Hex = hex.EncodeToString(sha1Sum[:])

    sha256Sum := sha256.Sum256(data)
    sha256Hex = hex.EncodeToString(sha256Sum[:])
    sha256Base64 = base64.StdEncoding.EncodeToString(sha256Sum[:])

    sha512Sum := sha512.Sum512(data)
    sha512Hex = hex.EncodeToString(sha512Sum[:])
    sha512Base64 = base64.StdEncoding.EncodeToString(sha512Sum[:])

    md5Sum := md5.Sum(data)
    md5Hex = hex.EncodeToString(md5Sum[:])

    return
}
```

Reply:
I decided to open a new PR for my suggested changes #142

Don't know how much more successful my PR will be than this one :|